backend/src/services/aws/ec2/ec2.service.ts

```diff
@@ -88,10 +89,21 @@ class EC2 {
     );
 
     const [instance] = data.Instances as Instance[];
-    const { InstanceId: instanceId } = instance;
+    const { InstanceId: instanceId, State: state } = instance;
+    const { Name: stateName } = state as InstanceState;
 
     await this.setInstanceName(instanceId as string, name);
 
+    return {
+      instanceId: instanceId as string,
+      state: stateName as string,
+    };
+  }
+
+  public async getRunningInstanceData(instanceId: string): Promise<{
+    hostname: string;
+    state: string;
```

Comment:
Passing state as a string is always a bad idea. It is the same as using string as a date everywhere in the app. In some places it can be 'running', in other - 'Running'. Things get even worse when you are trying to remove status. Imagine that we decided to drop 'active' status for an instance and replace it by 'running'. Only way to do this migration is to change all 'active' occurrences in code by 'running'. But suddenly It turns out that word 'active' used 1000 times in your app in completely different places 😨